torch/csrc/jit/tensorexpr/ir_simplifier.h

```diff
@@ -114,28 +112,25 @@ inline ExprPtr evaluateOp(ExprPtr v) {
 class Term : public ExprNode<Term> {
  public:
   template <class... Args>
-  // NOLINTNEXTLINE(cppcoreguidelines-pro-type-member-init)
   Term(HashProvider& hasher, ExprPtr s, Args... ts)
       : ExprNodeBase(promoteTypesVar(s, ts...)), scalar_(s), hasher_(hasher) {
     CHECK(s->isConstant());
     addComponent(ts...);
     sort();
   }
 
-  // NOLINTNEXTLINE(cppcoreguidelines-pro-type-member-init)
   Term(HashProvider& hasher, ExprPtr s, std::vector<ExprPtr> v)
       : ExprNodeBase(promoteTypesVec(s, v)),
         variables_(std::move(v)),
-        scalar_(s),
+        scalar_(std::move(s)),
         hasher_(hasher) {
     sort();
   }
 
   // Convenience constructor from a map of hash -> var, used when merging Terms.
-  // NOLINTNEXTLINE(cppcoreguidelines-pro-type-member-init)
   Term(
       HashProvider& hasher,
-      ExprPtr s,
+      const ExprPtr& s,
```

Comment:
shouldn't we just use `scalar_` for anything we want to reference in the ctor (and use std::move to move it into scalar_)?